src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -858,6 +858,19 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
             // of the script to evaluate. If -file comes before -command, it will
             // treat -command as an argument to the script...
 
+            bool? GetBoolValue(string arg)
+            {
+                if (arg.Equals("$true", StringComparison.OrdinalIgnoreCase))
```

Comment:
Isn't the use case:
```
powershell -file foo.ps1 -myswitch:$true
```

Not clear to me when we need `True` as I don't think we want to support:

```
powershell -file foo.ps1 -myswitch:true
```